analysis/bulk-deconvolution/exploratory-notebooks/quantiseq-tumor-genes.Rmd

```diff
@@ -0,0 +1,264 @@
+* `FCRLA`: cell surface glycoprotein, involved in inflammatory response
+* `SLC1A3`: glutamate transporter
+
+On the whole, while these genes which may be associated with (adult) tumors, they are not necessarily oncogenes or causatively linked to tumor progression.
+
+Since these genes were identified from an database of _adult_ tumor samples, it's not clear whether they have similar biological relevance to pediatric tumors. 
+The purpose of this notebook is to assess whether these genes indeed have aberrantly high expression values compared to the other signature genes in each of our bulk RNA-seq samples.
+
+## Confirm signature genes are present
+
+Since we use gene symbol identifiers here, we should confirm that all gene symbols in the `quanTIseq` reference are indeed present in our data. 
+For this, we'll check that the signature genes are all present in our mapping table which converted ensembl ids to gene symbols.
+
+```{r}
+map_table <- readr::read_tsv(
+  here::here("analysis", "bulk-deconvolution", "data", "reference", "ensembl_symbol.tsv"),
+  show_col_types = FALSE
+)
+map_symbols <- map_table$gene_symbol
+
+all(signature_genes %in% map_symbols)
+```
+
+We are indeed missing some! Which ones?
+
+```{r}
+signature_genes[which(!(signature_genes %in% map_symbols))]
+```
+
+It turns out the reason these are "missing" is because they have been deprecated since `quanTIseq` was written, or they are aliases for the official gene symbol:
+
+- `AKAP2` was deprecated and replaced with `PALM2AKAP2` (source: <https://www.ncbi.nlm.nih.gov/gene/11217>)
+- `FAM46C` is an alias for `TENT5C` (source: <https://www.ncbi.nlm.nih.gov/gene/54855>)
+- `GUCY1A3` is an alias for `GUCY1A1` (source: <https://www.ncbi.nlm.nih.gov/gene/2982>)
+
+Are these gene symbols present in our ensembl <-> symbol mapping table?
+
+```{r}
+updated_gene_names <- c("PALM2AKAP2", "TENT5C", "GUCY1A1")
+all(updated_gene_names %in% map_symbols)
+```
+
+Great, we have these other versions of the gene symbols!
+We'll directly replace them in the signature list here.
+
+```{r}
+signature_genes <- dplyr::case_match(
+  signature_genes, 
+  "AKAP2" ~ "PALM2AKAP2", 
+  "FAM46C" ~ "TENT5C", 
+  "GUCY1A3" ~ "GUCY1A1", 
+  .default = signature_genes
+)
+```
+
+When we actually run `quanTIseq`, we'll need to make sure that we have matching gene names between our TPM matrices and their signature.
+
+## Prepare the data
+
+We'll read in the TPM data from all projects into a single long data frame, retaining only the signature genes with an indicator column for which are the tumor genes.
+
+```{r}
+# Create data frame of TPM values for all samples
+tpm_df <- tpm_files |>
+  purrr::map(
+    \(x) {
+      # read matrix and convert to table
+      readr::read_rds(x) |>
+        as.data.frame() |>
+        tibble::rownames_to_column(var = "gene_symbol")
+      }
+  ) |> 
+  # Combine into a single data frame
+  purrr::reduce(dplyr::left_join, by = "gene_symbol") |>
```

Comment:
Just say I would probably have done this with `cbind` as we know the gene lists are the same... joining is slow!
```suggestion
  purrr::map(readr::read_rds) |> 
  # combine into a single matrix
  purrr::reduce(cbind) |>
  as.data.frame() |>
  tibble::rownames_to_column(var = "gene_symbol") |>
```